### PR TITLE
[Query enhancements] use status 503 if search strategy throws 500

### PR DESCRIPTION
**Pull request by joshuali925**
### Description

use status 503 if search strategy throws 500 to indicate downstream service failure

## Testing the changes

when opensearch is down, send a request to enhancements api:
before
```sh
❯ curl -X POST 'http://localhost:5601/api/enhancements/search/sql' -H 'Content-Type: application/json' -H 'osd-xsrf: osd-fetch' -d '{"query":{ "query": "b", "language": "language", "format": "format" }}'
{"statusCode":500,"error":"Internal Server Error","message":"An internal server error occurred."}%
```
after
```sh
❯ curl -X POST 'http://localhost:5601/api/enhancements/search/sql' -H 'Content-Type: application/json' -H 'osd-xsrf: osd-fetch' -d '{"query":{ "query": "b", "language": "language", "format": "format" }}'
{"statusCode":503,"error":"Service Unavailable","message":"Error: No Living connections"}%
```

## Changelog
- skip

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/query_enhancements/server/routes/index.ts [hunk 1]
```diff
@@ -91,8 +91,9 @@ export function defineSearchStrategyRouteProvider(logger: Logger, router: IRoute
           } catch (e) {
             error = err;
           }
+          const statusCode = error.status || err.status;
           return res.custom({
-            statusCode: error.status || err.status,
+            statusCode: statusCode === 500 ? 503 : statusCode || 503,
```
Comment (virajsanghvi) [inline src/plugins/query_enhancements/server/routes/index.ts:96]: can we validate this behavior in a test?

Comment (joshuali925) [reply src/plugins/query_enhancements/server/routes/index.ts]: added a basic test

**Comment by kavilla** [comment]
i think the test failures might be due to tests needing to be updated, or snapshots

**Comment by joshuali925** [comment]
failure seems irrelevant, they passed on rerun
```
Summary of all failing tests
 FAIL  src/plugins/data/public/query/query_string/query_string_manager.test.ts
  ● QueryStringManager › clearQueryHistory clears the query history

    expect(received).toHaveLength(expected)

    Expected length: 2
    Received length: 1
    Received array:  [{"id": "8ffe6413-d2bf-45ae-863b-0b7a80208bcd", "query": {"language": "sql", "query": "test query 2"}, "time": 1731698469371}]

      109 |     service.addToQueryHistory({ query: 'test query 1', language: 'sql' });
      110 |     service.addToQueryHistory({ query: 'test query 2', language: 'sql' });
    > 111 |     expect(service.getQueryHistory()).toHaveLength(2);
          |                                       ^
      112 |
      113 |     service.clearQueryHistory();
      114 |     expect(service.getQueryHistory()).toHaveLength(0);

      at Object.<anonymous> (src/plugins/data/public/query/query_string/query_string_manager.test.ts:111:39)

  ● QueryStringManager › addToQueryHistory adds query to history

    expect(received).toHaveLength(expected)

    Expected length: 1
    Received length: 2
    Received array:  [{"id": "4714cdd6-cd94-4736-8a88-919c873419d1", "query": {"language": "sql", "query": "test query"}, "time": 1731698469376}, {"id": "8ffe6413-d2bf-45ae-863b-0b7a80208bcd", "query": {"language": "sql", "query": "test query 2"}, "time": 1731698469371}]

      119 |     service.addToQueryHistory(query);
      120 |     const history = service.getQueryHistory();
    > 121 |     expect(history).toHaveLength(1);
          |                     ^
      122 |     expect(history[0]).toHaveProperty('query', query);
      123 |   });
      124 |

      at Object.<anonymous> (src/plugins/data/public/query/query_string/query_string_manager.test.ts:121:21)
```

**Review by kavilla — APPROVED** [review]

**Review by virajsanghvi — APPROVED** [review]

**Pull request merged**